src/main/java/blusunrize/immersiveengineering/common/EventHandler.java

```diff
@@ -312,20 +313,14 @@ public void onWorldTick(WorldTickEvent event)
 						itExplosion.remove();
 				}
 			}
-			synchronized (requestedBlockUpdates)
+			while (!requestedBlockUpdates.isEmpty())
 			{
-				while (!requestedBlockUpdates.isEmpty())
+				Pair<Integer, BlockPos> curr = requestedBlockUpdates.poll();
+				World w = DimensionManager.getWorld(curr.getLeft());
+				if(w!=null)
 				{
-					Pair<Integer, BlockPos> curr = requestedBlockUpdates.poll();
-					if(FMLCommonHandler.instance().getEffectiveSide()==Side.SERVER)

```

Comment:
I don't think it's a good idea to remove the side check here? It won't cause crashes, but it will cause unnecessary block updates on the client.

Reply:
By looking at the call hierachies I concluded that WorldTickEvent is only fired for server worlds. This may be unintentional and a bug in Forge, but Forge cannot change this without breaking assumptions like this one.

Use of `FMLCommonHandler#getEffectiveSide()` is discouraged due to the nature of how it works. In this case, where a World is available, `World#isRemote` should be used instead.

It's up to you. I can re-add the side check as a `World#isRemote` check, but it is strictly not necessary.

Reply:
I'll check whether it's called for client worlds when I'm home, if it isn't it should be fine like this.

Reply:
Yes, you're right. Good to know.